streamduck-daemon/src/main.rs

```diff
@@ -198,6 +203,37 @@ fn hide_console() {
     }
 }
 
+async fn check_for_dirty_configs(config: &Arc<Config>) {
+    let device_config = config.get_all_device_configs().await;
+
+    for lock in device_config {
+        let conf = lock.read().await;
+        let timediff = conf.commit_duration();
+        if conf.is_dirty() && timediff > Duration::from_secs(30) {
+            drop(conf);
+            let mut conf = lock.write().await;
+            // set clean state
+            conf.set_clean();
+            let serial = conf.serial.clone();
+            // clear the lock
+            drop(conf);
```

Comment:
Would had been so much better if instead of marking clean here, it'd be done in save_device_config method, so even if the config was saved by something else, autosave wouldn't do the useless thing of writing the config again